pkg/auth/common/login_handler.go

```diff
@@ -165,6 +178,54 @@ func (h *TanzuLoginHandler) getTokenWithRefreshToken() (*Token, error) {
 	}, nil
 }
 
+// Create or update the cert map entry for the issuer if necessary
+func (h *TanzuLoginHandler) updateCertMap() {
+	// explicitly provided cert info was successfully used to authenticate, so as a
+	// best-effort: save them in the cert map for convenience if necessary
+	if h.tlsSkipVerify || h.caCertData != "" {
+		u, err := url.Parse(h.issuer)
+		if err != nil {
+			fmt.Printf("Unable to parse issuer %s: %v\n", h.issuer, err)
+			return
+		}
+		host := u.Hostname()
+		if host == "" {
+			host = h.issuer
+		}
+
+		var cert *configtypes.Cert
+		found, _ := config.GetCert(host)
+
+		tlsSkipVerifyStr := strconv.FormatBool(h.tlsSkipVerify)
+		if found != nil {
+			if found.CACertData != h.caCertData || found.SkipCertVerify != tlsSkipVerifyStr {
+				cert = &configtypes.Cert{
+					Host:           host,
+					CACertData:     h.caCertData,
+					SkipCertVerify: tlsSkipVerifyStr,
+					Insecure:       found.Insecure,
+				}
+				if err = config.DeleteCert(host); err != nil {
```

Comment:
nit: Do we need to DeleteCert before SetCert or will SetCert update the existing one? I thought SetCert will update as well.

Reply:
I was wondering the same. The login of the cert add command performs an explicit delete first, which I understand now was to deal with when the skipverify field is are not explicitly set when calling SetCert. This is not a problem here, so I can remove it.